app/models/ZkKafka.scala

```diff
@@ -81,11 +82,14 @@ object ZkKafka {
       // For each partition, fetch the offsets.
       parts.asScala.map({ p =>
         val jsonState = zkClient.getData.forPath(makePath(applyBase(Seq(stormZkRoot, Some(root))) ++ Seq(Some(pts)) ++ Seq(Some(p))))
-        val state = Json.parse(jsonState)
-        val offset = (state \ "offset").as[Long]
-        val partition = (state \ "partition").as[Long]
-        (partition.toInt, offset)
-      })
+        tryParse(jsonState).map { state =>
```

Comment:
Question: I'm still feeling my way around all the `Option` stuff, but would this be equivalent?

```scala
tryParse(jsonState).flatMap { state =>
  val offset = (state \ "offset").as[Long]
  val partition = (state \ "partition").as[Long]
  Some(partition.toInt -> offset)
}
```

Seems like both end up converting an `Option[JsValue]` to `Option[Map]`. I don't know if there's any standards dictating which would be more idiomatic, but intuitively the `flatMap` seems simpler to me. (I may also just have the types mixed up or something though.)